src/compiler/scala/tools/nsc/settings/StandardScalaSettings.scala

```diff
@@ -52,7 +51,26 @@ trait StandardScalaSettings { _: MutableSettings =>
   val nowarn =         BooleanSetting ("-nowarn", "Generate no warnings.") withAbbreviation "--no-warnings" withPostSetHook { s => if (s) maxwarns.value = 0 }
   val optimise:        BooleanSetting // depends on post hook which mutates other settings
   val print =          BooleanSetting ("-print", "Print program with Scala-specific features removed.") withAbbreviation "--print"
-  val target =         ChoiceSetting  ("-target", "target", "Target platform for object files.", AllTargetVersions, "8") withPreSetHook normalizeTarget _ withAbbreviation "--target"
+  val release =
+    ChoiceSetting("-release", "release", "Compile for a version of the Java API and target class file.", AllTargetVersions, normalizeTarget(javaSpecVersion))
+      .withPostSetHook { setting =>
+        val current = setting.value.toInt
+        if (!isJavaAtLeast("9") && current > 8) errorFn.apply("-release is only supported on JVM 9 and higher")
+        if (target.valueSetByUser.map(_.toInt > current).getOrElse(false)) errorFn("-release cannot be less than -target")
+        //target.value = setting.value  // this would trigger deprecation
+      }
+      .withAbbreviation("--release")
+      .withAbbreviation("-java-output-version")
+  def releaseValue: Option[String] = release.valueSetByUser
+  val target =
+    ChoiceSetting("-target", "target", "Target platform for object files.", AllTargetVersions, "8")
+      .withPreSetHook(normalizeTarget)
+      .withPostSetHook { setting =>
+        if (releaseValue.map(_.toInt < setting.value.toInt).getOrElse(false)) errorFn("-release cannot be less than -target")
+      }
+      .withAbbreviation("--target")
```

Comment:
Scala 3 uses `-Xunchecked-java-output-version`, `-Xtarget` and `--Xtarget`. As the first variant is currently preferred I think we should just add `-Xunchecked-java-output-version` here for compatibility without unnecessary confusion with the other aliases

Reply:
I was reluctant to add the excessively long option names, but as aliases they don't spoil the `-help`.